src/Symfony/Component/AutoMapper/AutoMapper.php

```diff
@@ -0,0 +1,122 @@
+    {
+        $refl = new \ReflectionClass($object);
+
+        if (!$to) {
+            $to = $this->getAttribute($refl, MapTo::class, true)->to;
+        }
+
+        $arguments = [];
+        if (is_object($to)) {
+            $toRefl = new \ReflectionClass(get_class($to));
+            $mapped = $to;
+        } else {
+            $toRefl = new \ReflectionClass($to);
+            $mapped = $toRefl->newInstanceWithoutConstructor();
+        }
+
+        $constructor = $toRefl->getConstructor();
+        foreach ($constructor?->getParameters() ?? [] as $parameter) {
+            $arguments[$parameter->getName()] = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+        }
+
+        foreach ($refl->getProperties() as $property) {
+            $propertyName = $property->getName();
+            $mapTo = $this->getAttribute($property, MapTo::class)?->to ?? $propertyName;
+            if (!$toRefl->hasProperty($mapTo)) {
+                continue;
+            }
+
+            $value = $this->propertyAccessor ? $this->propertyAccessor->getValue($object, $propertyName) : $object->{$propertyName};
+            $mapIf = $this->getCallable($this->getAttribute($property, MapIf::class)?->if);
+            if (is_callable($mapIf) && false === $this->call($mapIf, $value, $object)) {
+                continue;
+            }
+
+            $mapWith = $this->getCallable($this->getAttribute($property, MapWith::class)?->with);
+            if (is_callable($mapWith)) {
+                $value = $this->call($mapWith, $value, $object);
+            }
+
+            if (is_object($value) && $to = $this->getAttribute(new \ReflectionClass(get_class($value)), MapTo::class)?->to) {
+                $value = $this->map($value, $to);
+            }
+
+            if (array_key_exists($mapTo, $arguments)) {
+                $arguments[$mapTo] = $value;
+            } else {
+                $this->propertyAccessor ? $this->propertyAccessor->setValue($mapped, $mapTo, $value) : ($mapped->{$mapTo} = $value);
+            }
+        }
+
+        $constructor->invokeArgs($mapped, $arguments);
+
+        return $mapped;
+    }
+
+    private function call(callable $fn, mixed $value, object $object): mixed
+    {
+        $refl = new \ReflectionFunction(\Closure::fromCallable($fn));
+        $withParameters = $refl->getParameters();
+        $withArgs = [$value];
+
+        // Let's not send object if we don't need to, gives the ability to call native functions
+        foreach ($withParameters as $parameter) {
+            if ($parameter->getName() === 'object') {
+                $withArgs['object'] = $object;
+                break;
+            }
+        }
+
+        return call_user_func_array($fn, $withArgs);
+    }
+
+    private function getCallable(string|callable|null $fn): ?callable
+    {
+        if ($this->serviceLocator && is_string($fn) && $this->serviceLocator->has($fn)) {
```

Comment:
```suggestion
        if ($this->serviceLocator && is_string($fn)) {
```

Let's `get()` throwing if it's a string that isn't an existing service name.

Reply:
I'd like to keep that as it gives me less flexibility (use native php methods with a string)